base_comment_template/models/base_comment_template.py

```diff
@@ -15,69 +16,47 @@ class CommentTemplate(models.AbstractModel):
-                ("partner_ids", "in", partner_id),
-            ]
-            lang = self.env["res.partner"].browse(partner_id).lang
-        if company_id and "company_id" in self._fields:
-            if partner_id and "partner_id" in self._fields:
-                default_dom.insert(-3, "&")
-            default_dom += [
-                "|",
-                ("company_id", "=", company_id),
-                ("company_id", "=", False),
-            ]
-        templates = self.env["base.comment.template"].search(
-            default_dom, order="priority"
-        )
-        if lang:
-            templates = templates.with_context({"lang": lang})
-        return templates
+    comment_template_ids = fields.Many2many(
+        compute="_compute_comment_template_ids",
+        comodel_name="base.comment.template",
+        string="Comment Template",
+        store=True,
+        readonly=False,
+        domain=lambda self: [("model_ids.model", "=", self._name)],
+    )
 
-    def get_comment_template(
-        self, position="before_lines", company_id=False, partner_id=False
-    ):
-        """ Method that is called from report xml and is returning the
-            position template as a html if exists
-        """
-        self.ensure_one()
-        templates = self.get_comment_template_records(
-            position=position, company_id=company_id, partner_id=partner_id
-        )
-        template = False
-        if templates:
-            for templ in templates:
-                if self in self.search(safe_eval(templ.domain or "[]")):
-                    template = templ
-                    break
-        if not template:
-            return ""
-        return self.env["mail.template"]._render_template(
-            template.text, self._name, self.id, post_process=True
-        )
+    @api.depends(_comment_template_partner_field_name)
+    def _compute_comment_template_ids(self):
+        for record in self:
+            partner = record[self._comment_template_partner_field_name]
+            record.comment_template_ids = [(5,)]
+            templates = self.env["base.comment.template"].search(
+                [
+                    ("id", "in", partner.base_comment_template_ids.ids),
+                    ("model_ids.model", "=", self._name),
+                ]
+            )
+            for template in templates:
+                if self in self.search(safe_eval(template.domain or "[]")):
+                    record.comment_template_ids = [(4, template.id)]
+
+    @api.depends("name", "position")
+    def name_get(self):
+        result = []
+        for record in self:
+            name = "{} ({})".format(record.name, record.position)
+            result.append((record.id, name))
+        return result
 
 
 class BaseCommentTemplate(models.Model):
     """Comment templates printed on reports"""
 
     _name = "base.comment.template"
     _description = "Comments Template"
+    _order = "priority"
```

Comment:
This is usually called `sequence`. The existing field is already at the view with the handle widget?

Reply:
It's a good moment to rename field. I have changed name field to `sequence` and have added migration script.